drivers/phoenixcontact_modbus.c

```diff
@@ -100,23 +110,17 @@ void upsdrv_initinfo(void)
 	/* upsh.instcmd = instcmd; */
 	/* upsh.setvar = setvar; */
 
-	mrir(modbus_ctx, 0x0004, 1, &FWVersion);
-	dstate_setinfo("ups.firmware", "%" PRIu16, FWVersion);
-
-	mrir(modbus_ctx, 0x0005, 1, &PartNumber1);
-	mrir(modbus_ctx, 0x0006, 1, &PartNumber2);
-	mrir(modbus_ctx, 0x0007, 1, &PartNumber3);
-	mrir(modbus_ctx, 0x0008, 1, &PartNumber4);
+	mrir(modbus_ctx, 0x0005, 4, &tab_reg);
 
 	/*	Method provided from Phoenix Conatct to establish the UPS model:
 		Read registers from 0x0005 to 0x0008 and "concatenate" them with the order
 		0x0008 0x0007 0x0006 0x0005 in hex form, convert the obtained number from hex to dec.
 		The first 7 most significant digits of the number in dec form are the part number of
 		the UPS.*/
 
-	PartNumber = (PartNumber4 * 65536) + PartNumber3;
-	PartNumber = (PartNumber * 65536) + PartNumber2;
-	PartNumber = (PartNumber * 65536) + PartNumber1;
+	PartNumber = (tab_reg[3] * 65536) + tab_reg[2];
+	PartNumber = (PartNumber * 65536) + tab_reg[1];
+	PartNumber = (PartNumber * 65536) + tab_reg[0];
```

Comment:
Sanity check: are you sure maths are not truncated here by a 16-bit number?

What does this multiplication achieve? (A bit shift would be more idiomatic)